src/mlpack/methods/hoeffding_trees/hoeffding_tree_impl.hpp

```diff
@@ -457,49 +440,35 @@ void HoeffdingTree<
          const arma::Row<size_t>& labels,
          const size_t numClasses,
          const bool batchTraining,
-         const double successProbability)
+         const std::optional<double> successProbability,
+         const std::optional<size_t> maxSamples,
+         const std::optional<size_t> checkInterval)
 {
-  Train(data, labels, numClasses, batchTraining, successProbability,
-      this->maxSamples, this->checkInterval, this->minSamples);
-}
 
-template<typename FitnessFunction,
-         template<typename> class NumericSplitType,
-         template<typename> class CategoricalSplitType>
-template<typename MatType>
-void HoeffdingTree<
-    FitnessFunction,
-    NumericSplitType,
-    CategoricalSplitType
->::Train(const MatType& data,
-         const arma::Row<size_t>& labels,
-         const size_t numClasses,
-         const bool batchTraining,
-         const double successProbability,
-         const size_t maxSamples)
-{
-  Train(data, labels, numClasses, batchTraining, successProbability, maxSamples,
-      this->checkInterval, this->minSamples);
-}
-
-template<typename FitnessFunction,
-         template<typename> class NumericSplitType,
-         template<typename> class CategoricalSplitType>
-template<typename MatType>
-void HoeffdingTree<
-    FitnessFunction,
-    NumericSplitType,
-    CategoricalSplitType
->::Train(const MatType& data,
-         const arma::Row<size_t>& labels,
-         const size_t numClasses,
-         const bool batchTraining,
-         const double successProbability,
-         const size_t maxSamples,
-         const size_t checkInterval)
-{
-  Train(data, labels, numClasses, batchTraining, successProbability, maxSamples,
-      checkInterval, this->minSamples);
+  if (successProbability.has_value())
```

Comment:
@shrit I think the test failure is the same issue here, this function just needs to be adapted to use the newer logic approach.

Reply:
Did I forget one of these? I look into it in a while